install.sh

```diff
@@ -0,0 +1,322 @@
+        exit 1
+    fi
+
+    log_info "Installing to $install_dir..."
+    mkdir -p "$install_dir"
+    cp "$binary_path" "$install_dir/"
+    chmod +x "$install_dir/$BINARY_NAME"
+
+    # Clean up
+    rm -rf "$temp_dir"
+
+    log_success "$BINARY_NAME installed successfully!"
+}
+
+# Main installation process
+main() {
+    log_info "Starting container-use installation..."
+
+    check_dependencies
+
+    local os=$(detect_os)
+    local arch=$(detect_arch)
+    log_info "Detected platform: $os/$arch"
+
+    local version=$(get_latest_version)
+    if [ -z "$version" ]; then
+        log_error "Failed to get latest release version"
+        exit 1
+    fi
+    log_info "Latest version: $version"
+
+    INSTALL_DIR=$(find_install_dir)
+    log_info "Installation directory: $INSTALL_DIR"
+
+    check_existing_cu
+
+    download_and_install "$os" "$arch" "$version" "$INSTALL_DIR"
+
+    # Check if install directory is in PATH
+    if ! echo "$PATH" | grep -q "$INSTALL_DIR"; then
+        log_warning "Installation directory $INSTALL_DIR is not in your PATH"
+        log_info "Add this to your shell profile (.bashrc, .zshrc, etc.):"
+        echo "    export PATH=\"$INSTALL_DIR:\$PATH\""
+        log_info "Then restart your terminal or run: source ~/.bashrc (or your shell's config file)"
+    fi
+
+    # Verify installation
+    if [ -x "$INSTALL_DIR/$BINARY_NAME" ]; then
+        log_success "Installation complete!"
+
+        # Check if the correct cu command is being found in PATH
+        local found_binary=$(command -v "$BINARY_NAME" 2>/dev/null || echo "")
+
+        if [ "$found_binary" = "$INSTALL_DIR/$BINARY_NAME" ]; then
+            log_success "$BINARY_NAME is ready to use!"
+        elif [ -n "$found_binary" ]; then
+            # Some other cu command is being found
+            local help_output=$("$found_binary" --help 2>&1 || true)
+            if echo "$help_output" | grep -q "Taylor UUCP"; then
+                log_error "Detected Taylor UUCP 'cu' command instead of container-use"
+                log_info "The system 'cu' command at $found_binary is taking precedence"
+                log_info "Try running: $INSTALL_DIR/$BINARY_NAME --help"
+                log_info "Or run 'hash -r' and try again"
+                log_info "Or add $INSTALL_DIR to the beginning of your PATH"
+                exit 1
+            else
+                log_warning "Different 'cu' command found at $found_binary"
+                log_info "Try running: $INSTALL_DIR/$BINARY_NAME --help"
+                log_info "Or run 'hash -r' and try again"
+                log_info "Or add $INSTALL_DIR to the beginning of your PATH"
+            fi
+        else
+            log_warning "You may need to restart your terminal or update your PATH"
+        fi
+
+        log_info "Run '$BINARY_NAME --help' to get started"
+    else
+        log_error "Installation verification failed"
+        exit 1
+    fi
+}
+
+# Handle command line arguments
+case "${1:-}" in
```

Comment:
this should be part of main to handle the partial download of the script being piped to sh. The only execution should be `main "$@"`